packages/utilities/fast-colors/src/color-converters.ts

```diff
@@ -117,7 +117,7 @@ function calcRgbOverlay(
 export function calculateOverlayColor(
     rgbMatch: ColorRGBA64,
     rgbBackground: ColorRGBA64,
-    rgbOverlay: ColorRGBA64 = null
+    rgbOverlay: ColorRGBA64 = null!
```

Comment:
I don't think this is the right fix. It should accept a null type and default to null

Reply:
`
export function calculateOverlayColor(
    rgbMatch: ColorRGBA64,
    rgbBackground: ColorRGBA64,
    rgbOverlay: ColorRGBA64 = null!
): ColorRGBA64 {
    let alpha: number = 0;
    let overlay: ColorRGBA64 = rgbOverlay;

    if (overlay !== null) {
        alpha = calcRgbOverlay(rgbMatch, rgbBackground, overlay);
    } else {
        overlay = new ColorRGBA64(0, 0, 0, 1);
        alpha = calcRgbOverlay(rgbMatch, rgbBackground, overlay);
        if (alpha <= 0) {
            overlay = new ColorRGBA64(1, 1, 1, 1);
            alpha = calcRgbOverlay(rgbMatch, rgbBackground, overlay);
        }
    }
    alpha = Math.round(alpha * 1000) / 1000;

    return new ColorRGBA64(overlay.r, overlay.g, overlay.b, alpha);
}
`

Reply:
`
   export function calculateOverlayColor(
    rgbMatch: ColorRGBA64,
    rgbBackground: ColorRGBA64,
    rgbOverlay: ColorRGBA64 = null!
   ): ColorRGBA64 {
    let alpha: number = 0;
    let overlay: ColorRGBA64 = rgbOverlay;

    if (overlay !== null) {
        alpha = calcRgbOverlay(rgbMatch, rgbBackground, overlay);
    } else {
        overlay = new ColorRGBA64(0, 0, 0, 1);
        alpha = calcRgbOverlay(rgbMatch, rgbBackground, overlay);
        if (alpha <= 0) {
            overlay = new ColorRGBA64(1, 1, 1, 1);
            alpha = calcRgbOverlay(rgbMatch, rgbBackground, overlay);
        }
    }
    alpha = Math.round(alpha * 1000) / 1000;

    return new ColorRGBA64(overlay.r, overlay.g, overlay.b, alpha);
   }
`

I have found in the function definition is it is already checking for null values and performing operations based on the values. Let me know whats on your mind. I am open to your thoughts.

Reply:
`
   export function calculateOverlayColor(
    rgbMatch: ColorRGBA64,
    rgbBackground: ColorRGBA64,
    rgbOverlay: ColorRGBA64 = null!
   ): ColorRGBA64 {
    let alpha: number = 0;
    let overlay: ColorRGBA64 = rgbOverlay;

    if (overlay !== null) {
        alpha = calcRgbOverlay(rgbMatch, rgbBackground, overlay);
    } else {
        overlay = new ColorRGBA64(0, 0, 0, 1);
        alpha = calcRgbOverlay(rgbMatch, rgbBackground, overlay);
        if (alpha <= 0) {
            overlay = new ColorRGBA64(1, 1, 1, 1);
            alpha = calcRgbOverlay(rgbMatch, rgbBackground, overlay);
        }
    }
    alpha = Math.round(alpha * 1000) / 1000;

    return new ColorRGBA64(overlay.r, overlay.g, overlay.b, alpha);
   }
`

I have found in the function definition is it is already checking for null values and performing operations based on the values. Let me know what's on your mind. I am open to your thoughts.

Reply:
@nicholasrice shall we come to a discussion to close this issue?